custom/intrahealth/sqldata.py

```diff
@@ -183,58 +184,46 @@ def group_by(self):
         return group_by
 
     @property
+    @memoized
+    def products(self):
+        return list(SQLProduct.objects.filter(domain=self.config['domain'], is_archived=False))
+
+    @property
     def rows(self):
+        products = self.products
+        values = {
+            product.product_id: [0, 0, 0]
+            for product in products
+        }
+
+        rows = super(DispDesProducts, self).rows
+        for row in rows:
+            product_id = row[0]
+            values[product_id] = [row[1], row[2], "%d%%" % (100 * row[2]['html'] / (row[1]['html'] or 1))]
```

Comment:
Might make sense to override the `rows` property to use a `DictDataFormat` which would make this much easier to read. I think you can just call `self.get_data()` to get a list of dicts as the rows